packages/components/src/color-palette/index.native.js

```diff
@@ -77,17 +68,10 @@ function ColorPalette( {
 	const customIndicatorColor = isGradientSegment
 		? activeColor
 		: customSwatchGradients;
-
-	useEffect( () => {
-		setShouldShowCustomIndicator(
-			shouldShowCustomIndicatorOption &&
-				( ! isGradientSegment || isCustomGradientColor )
-		);
-	}, [
-		shouldShowCustomIndicatorOption,
-		isGradientSegment,
-		isCustomGradientColor,
-	] );
```

Comment:
I think reverting #28740 is likely a fine approach, but I am curious as to why it was not working. 

I haven't explored it fully, but I wonder if the dependency array is not inclusive enough. It included `isCustomGradientColor`, but that value is computed with `isSelectedCustom`, which relies upon the values `activeColor`, `colors`, `isGradientSegment`, and `isGradientColor`. 

I would _think_ React would appropriately "watch" for changes to all those sub-dependencies of `isSelectedCustom`, but I wonder if maybe that is not occurring. Maybe we need to declare all of those values as dependencies for this hook?

Reply:
Thank you for your feedback and reviewing this so quickly @dcalhoun 🙇 
I was able to reproduce both the reported issues.

> Maybe we need to declare all of those values as dependencies for this hook?

I will revisit this with this approach and report back 👍

Reply:
Hey @dcalhoun 👋 
Thank you once more for the tip on this. Passing all the values and doing the calculation within the hook (85ff940ceff58da9436fb172dab2f3e25d99b94d) only fixes the [disappearing custom gradient](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3104#issuecomment-820581892) and not the original [issue](https://github.com/WordPress/gutenberg/issues/30878). Tbh I'm still a bit confused on why this happens only only compiled builds. I'll continue the investigation and get back 😕

Reply:
Interesting. https://github.com/WordPress/gutenberg/issues/30878 seems like either a style issue (i.e. some flex styles are conflicting or causing rendering issues), a calculation issue (i.e. there is somewhere else where we do not always re-calculate when we should), or a race condition (e.g. a computation occurs during layout and finishes at different times). 

I might start by placing "debug styles" on each element to see _which_ element is the offending item "cutting off" the children. Specifically, I know `ScrollView` contains multiple elements, a container and a content container. Which one is rendering too skinny?

Relatedly, the broken scroll-to-end for custom colors/gradients (#3104) is likely caused by the same issue where we compute the status within the hook, but do not have a dependency declared on the sub-dependency values.

https://github.com/WordPress/gutenberg/blob/85ff940ceff58da9436fb172dab2f3e25d99b94d/packages/components/src/color-palette/index.native.js#L102-L110

Applying a similar change as https://github.com/WordPress/gutenberg/commit/85ff940ceff58da9436fb172dab2f3e25d99b94d would likely resolve the scroll issue.